src/OsiXpr/OsiXprSolverInterface.cpp

```diff
@@ -69,18 +69,24 @@ static void reporterror(const char *fname, int iline, int ierr)
 
 void OsiXprSolverInterface::initialSolve()
 {
+  const char * args = "l";
+  bool takeHint;
+  OsiHintStrength strength;
+  getHintParam(OsiDoDualInInitial, takeHint, strength);
+  if (strength != OsiHintIgnore) {
+    args = takeHint ? "dl" : "pl";
+  }
 
   freeSolution();
 
 #if XPVERSION <= 20
   if (objsense_ == 1.0) {
-    XPRS_CHECKED(XPRSminim, (prob_, "l"));
+    XPRS_CHECKED(XPRSminim, (prob_, args));
   } else if (objsense_ == -1.0) {
-    XPRS_CHECKED(XPRSmaxim, (prob_, "l"));
+    XPRS_CHECKED(XPRSmaxim, (prob_, args));
   }
 #else
-
-  XPRS_CHECKED(XPRSlpoptimize, (prob_, "l"));
+  XPRS_CHECKED(XPRSmipoptimize, (prob_, args));
```

Comment:
This doesn't look right. `initialSolve()` (and `resolve()`) are for solving an LP, or the LP relaxation of a MIP, only. `branchAndBound()` is for solving a MIP.

Reply:
I was a little fast. Should maybe be
```suggestion
  XPRS_CHECKED(XPRSlpoptimize, (prob_, args));
```

and `l` should not be in `args` for LPs. It solves the LP relaxaion only when calling `XPRSmipoptimize` that is why it worked.

Reply:
Probably.